### ISSUE TITLE
Wrong test status icon after `completed` event

### ISSUE THREAD

**Issue by matepek** [hbenl/vscode-test-explorer#32]
Hey,

Test tree:
```
all 
 |-- suite1
 |       |--  s1test1
 |       |--  s1test2 
 |       \__  s1test3
 |
 |-- suite2
 |       |--  s2test1
 |       |--  s2test2
 |       \__  s2test3
 |
 |-- suite3
 |       |--  s3test1
 |       |--  s3test2
 |       \__  s3test3
```

I'm sending events like this:
```
	  { type: 'started', tests: ['all'] }
	  { type: 'suite', suite: 'suite1', state: 'running' }
	  { type: 'test', test: 's1test1', state: 'running' }
	  { type: 'test', test: 's1test1', state: 'passed' }
          { type: 'test', test: 's1test2', state: 'running' }
	  { type: 'test', test: 's1test2', state: 'failed' }
	  { type: 'suite', suite: 'suite1', state: 'completed' }
	  { type: 'suite', suite: 'suite2', state: 'running' }
	  { type: 'test', test: 's2test1', state: 'running' }
	  { type: 'test', test: 's2test1', state: 'passed' }
          { type: 'test', test: 's2test2', state: 'running' }
	  { type: 'test', test: 's2test2', state: 'failed' }
          { type: 'suite', suite: 'suite2', state: 'completed' }
          { type: 'suite', suite: 'suite3', state: 'running' }
	  { type: 'test', test: 's3test1', state: 'running' }
	  { type: 'test', test: 's3test1', state: 'passed' }
          { type: 'test', test: 's3test2', state: 'running' }
	  { type: 'test', test: 's3test2', state: 'failed' }
	  { type: 'suite', suite: 'suite3', state: 'completed' }
	  { type: 'finished' }
```

**Symptom**: The status icons of `suite1` and `suite2` remain "red play" (*running and failed*) until the `finished` event. It should be a "red X" (*finished and failed*)

I think the problem is that there is no event about `s1test3`, `s2test3` and `s3test3`.

Real example: My executable crashes during a test (`test3->1` in the example bellow), I detect it, send a failed event, but no events for the rest of the tests in the executable.

<img width="216" alt="screenshot 2019-01-06 at 9 19 30" src="https://user-images.githubusercontent.com/16504598/50733936-bfac0880-1196-11e9-93d4-f8b5fa04a2a5.png">

(Also: `test4`'s icon isn't right, but I assume it's a vscode drawing bug. It happens a lot on OSX 10.14.2. Appears after mouseover / mouse hover.)

**Context**:
- Test Explorer UI version: 2.3.1
- OS: OSX 10.14.2

**Comment by hbenl** [comment]
> I think the problem is that there is no event about s1test3, s2test3 and s3test3.

Yes, exactly. Currently I only "clean up" the test states when I receive the `{ type: 'finished' }` event. I could also do it when I receive a `{ type: 'suite', suite: '...', state: 'completed' }` event, but you'd also have to send that event for the crashed executable. The reason I need this event and cannot assume that a suite is completed when I receive a `{ type: 'suite', suite: '...', state: 'running' }` for a different (unrelated) suite is that I want the Test Explorer to work with adapters that run multiple suites in parallel. But I guess sending the `{ type: 'suite', suite: '...', state: 'completed' }` event shouldn't be a problem, right?

**Comment by matepek** [comment]
I send the “completed” event after I detect the crash, and my adapter can run multiple executables at the same time, so I’m using your feature. 

Agree, that would solve it. 
Thanks,

**Comment by hbenl** [comment]
I have just published version 2.3.2 containing the fix. Let me know if it works as expected.

**Comment by matepek** [comment]
checked: neat
